src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -1,60 +1,103 @@
 package co.inventorsoft.academy.collections.model;
 
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Set;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    private HashSet<T> range;
+
+    public Range(){
+        range = new HashSet<T>();
+    }
+
     public int size() {
-        return 0;
+        return range.size();
     }
 
     public boolean isEmpty() {
-        return false;
+        return range.isEmpty();
     }
 
     public boolean contains(Object o) {
-        return false;
+        return range.contains(o);
     }
 
     public Iterator<T> iterator() {
-        return null;
+        return range.iterator();
     }
 
     public Object[] toArray() {
-        return new Object[0];
+        return range.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+        return (T1[]) range.toArray(a);
     }
 
     public boolean add(T t) {
-        return false;
+        return range.add(t);
     }
 
     public boolean remove(Object o) {
-        return false;
+        return range.remove(o);
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return range.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+        return range.addAll(c);
     }
 
     public boolean retainAll(Collection<?> c) {
-        return false;
+        return range.retainAll(c);
     }
 
     public boolean removeAll(Collection<?> c) {
-        return false;
+        return range.removeAll(c);
     }
 
     public void clear() {
+        range.clear();
+    }
+
+
+    public static Range<Integer> of( int element1, int element2){
+        Range<Integer> rInt = new Range<>();
+        if(element1 == element2){
+            return rInt;
+        }
+        for(int i=element1;i<=element2;i++){
+            rInt.add(i);
+        }
+        return rInt;
+    }
+
+    public static Range<Float> of( float element1, float element2){
```

Comment:
Think about how to call method **Range.of(T start, T end, Function<T, T> function)** instead of writing code here.